File: pulse-go/internal/pulse/engine/value.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package engine

type value = int

const (
	// Piece values as defined by Larry Kaufman
	pawnValue   value = 100
	knightValue value = 325
	bishopValue value = 325
	rookValue   value = 500
	queenValue  value = 975
	kingValue   value = 20000

	infinite           value = 200000
	checkmate          value = 100000
	checkmateThreshold value = checkmate - MaxPly
	draw               value = 0

	noValue value = 300000
)

File: pulse-kotlin/src/commonMain/kotlin/com/fluxchess/pulse/kotlin/engine/Value.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package com.fluxchess.pulse.kotlin.engine

internal typealias Value = Int

// Piece values as defined by Larry Kaufman
internal const val PAWN_VALUE: Value = 100
internal const val KNIGHT_VALUE: Value = 325
internal const val BISHOP_VALUE: Value = 325
internal const val ROOK_VALUE: Value = 500
internal const val QUEEN_VALUE: Value = 975
internal const val KING_VALUE: Value = 20000

internal const val INFINITE: Value = 200000
internal const val CHECKMATE: Value = 100000
internal const val CHECKMATE_THRESHOLD: Value = CHECKMATE - MAX_PLY
internal const val DRAW: Value = 0

internal const val NO_VALUE: Value = 300000
